### PR TITLE
Explain with more detail the alerts behavior

### PR DESCRIPTION
**Pull request by pereyra-m**
## Description

This PR updates the description of the vulnerability scanner module so the issue can understand when to expect an alert and when not.
I've also made a reference to the PoC to show an example of the vulnerabilities inventory.

The alert generation schema is proposed here as a draft. When it's ready, we can convert with this same style https://documentation.wazuh.com/current/development/wazuh-cluster.html

Diagrama en blanco (1)

**Note:** this is the second version of the diagram

### Docs building
- [x] Compiles without warnings.

### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.

### CODE REVIEW

File: source/proof-of-concept-guide/poc-vulnerability-detection.rst [hunk 1]
```diff
@@ -61,11 +61,15 @@ Test the configuration
 
 #. Remove the Vim package to fix the vulnerability. Wait for syscollector to run a new scan.
 
-Visualize the alerts
---------------------
+If the syscollector scan has finished but you don't find the changes reflected in the Wazuh dashboard, it might be the case that the indexation cool down period for that agent is preventing the indexation. This 30 minutes time window can't be changed.
```
Comment (sebasfalcone) [inline source/proof-of-concept-guide/poc-vulnerability-detection.rst:64]: Maybe we could mention that this feature is named "abuseControl" so the user can make that connection with the logs

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 2]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
+For Microsoft Windows systems and specific Microsoft products, the Vulnerability Detection module has the :ref:`wodle_syscollector_hotfixes` option in the :ref:`syscollector <system_inventory>` settings on the Wazuh agent. Using this option enables the module to detect packages that the user has patched. When the Vulnerability Detection module detects a patch, it uses the information provided by Microsoft to decide if the patch has resolved the CVEs. Then, it removes them from the list of vulnerabilities.
+
+.. _vuln_det_alert_generation:
 
 Alert generation
 ----------------
 
-The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when users fix identified vulnerabilities. You can see this workflow in the diagram below.
+The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when users fix identified vulnerabilities.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:34]: "or when users fix identified vulnerabilities." Not necessarily the only case, maybe an automatic update

Maybe we can say something along the lines of "when a vulnerability is fixed due to a package update, removal or system upgrade"

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 3]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
+
+ You can see this workflow in the diagram below.
 
 .. thumbnail:: /images/manual/vuln-detector/vuln-detector-workflow.png
    :title: Vulnerability detection workflow
    :alt: Vulnerability detection workflow
    :align: center
    :width: 80%
 
+Considering all the possible situations, the most reliable way to determine the current vulnerabilities of the environment is to query the vulnerability inventory.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:53]: Best part of all

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 4]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
+For Microsoft Windows systems and specific Microsoft products, the Vulnerability Detection module has the :ref:`wodle_syscollector_hotfixes` option in the :ref:`syscollector <system_inventory>` settings on the Wazuh agent. Using this option enables the module to detect packages that the user has patched. When the Vulnerability Detection module detects a patch, it uses the information provided by Microsoft to decide if the patch has resolved the CVEs. Then, it removes them from the list of vulnerabilities.
+
+.. _vuln_det_alert_generation:
 
 Alert generation
 ----------------
 
-The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when users fix identified vulnerabilities. You can see this workflow in the diagram below.
+The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when users fix identified vulnerabilities.
+However, to enhance the reliability of Wazuh platform, the module doesn't always generate alerts due to a vulnerability detected or solved. An alert means that something has just happened, and that isn't always the case:
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:35]: "due to a vulnerability detected or solved." I didn't understood this part

**Review by sebasfalcone — COMMENTED** [review]
Graph LGTM! Some minor changes requested

File: source/proof-of-concept-guide/poc-vulnerability-detection.rst (hunk 1 above)
Comment (pereyra-m) [reply source/proof-of-concept-guide/poc-vulnerability-detection.rst:64]: I've updated the sentence.

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst (hunk 2 above)
Comment (pereyra-m) [reply source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:34]: This is more clear and its similar to the text of the alerts. I'll update it.

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst (hunk 4 above)
Comment (pereyra-m) [reply source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:35]: I've updated the sentence.

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 5]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
 
 Subsequently, we merge this content, uploading the compiled documents to a cloud server. And finally, we publish these documents to our CTI API.
 
-To update the vulnerabilities information on the Wazuh server, the vulnerability detection module queries the CTI API or an offline local repository. It obtains new documents and any resulting difference with old ones. Then the module publishes new content to a channel notifying subscribers.
+To update the vulnerabilities information on the Wazuh server, the vulnerability detection module queries the CTI API or an offline local repository. It obtains new documents and any resulting difference with old ones.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:21]:
```suggestion
To update the vulnerabilities information on the Wazuh server, the vulnerability detection module queries the CTI API or an offline local repository. It retrieves new documents and compares them to the existing ones to identify any differences.
```

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 6]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
+Then, the module scans the software inventory of the endpoints using the latest available vulnerabilities information. The detection process looks for vulnerable packages in the inventory databases. These inventories are unique to each agent.
 
-A package is labeled as vulnerable when its version matches those within the affected range of a CVE. Alerts show the results, and the module stores the findings in a per-agent vulnerability inventory. This inventory contains the current state of every agent and includes vulnerabilities that have been detected and not resolved. Users can :doc:`query the inventory </user-manual/capabilities/system-inventory/index>` to check for alerts and vulnerability information.
+A package is labeled as vulnerable when its version matches those within the affected range of a CVE. Alerts show the results, and the module stores the findings in a general vulnerability inventory that can be filtered per-agent. This inventory contains the current state of every agent and only includes vulnerabilities that have been detected and not resolved. Users can :ref:`query the inventory <vuln_det_poc_alert_visualize>` to check for alerts and vulnerability information.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:25]:
```suggestion
A package is marked as vulnerable when its version falls within the affected range of a CVE. The module generates alerts to display the results, and the findings are stored in a general vulnerability inventory. This inventory is filterable by agent and contains only unresolved vulnerabilities. Users can :ref:`query the inventory <vuln_det_poc_alert_visualize>` to review alerts and vulnerability details.
```

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 7]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
 
 Alert generation
 ----------------
 
-The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when users fix identified vulnerabilities. You can see this workflow in the diagram below.
+The Vulnerability Detection module generates alerts when it detects new vulnerabilities or when a vulnerability is fixed due to a package update, removal or system upgrade.
+However, to enhance the reliability of Wazuh platform, the conditions just mentioned are required but not enough for the alert generation. An alert means that something has just happened, and that isn't always the case:
+
+-  OS alerts: they won't be generated in the initial scan but in the subsequent ones. If the agent is syncing with the manager by the first time, it isn't considered that the OS version has changed or a patch has been applied recently.
+-  Package alerts: the package installation/removal that caused the vulnerabilities inventory change will trigger an alert if the agent detects the event in a normal :ref:`syscollector <system_inventory>` scan. The changes that occur while the agent is stopped or that are forced to be reported with a restart won't trigger an alert.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:38]:
```suggestion
OS alerts: These are not generated during the initial scan, but in subsequent scans. If the agent is syncing with the manager for the first time, it is not considered that the OS version has changed or that a patch was recently applied.
Package alerts: Alerts are triggered when the installation or removal of a package results in a change in the vulnerabilities inventory. However, this occurs only when the agent detects the event during a regular :ref:`syscollector <system_inventory>` scan. Changes made while the agent is stopped or that are forced to be reported through a restart do not generate alerts.
```

File: source/user-manual/capabilities/vulnerability-detection/how-it-works.rst [hunk 8]
```diff
@@ -18,25 +18,40 @@ We maintain the integrity of our vulnerability data by doing the following addit
...
+
+There are other considerations to take into account about the alert generation:
+
+-  Cluster environment: if an agent changes the node it's reporting to, its inventory will be synced with the new node but no alerts will be generated in this first sync.
+-  Content update: the changes in the content trigger a re-scan of all agents to keep the results up to date, but no alerts are generated.
+
+ You can see this workflow in the diagram below.
```
Comment (sebasfalcone) [inline source/user-manual/capabilities/vulnerability-detection/how-it-works.rst:45]:
```suggestion
Other factors to consider regarding alert generation include:

-  Cluster environment: If an agent switches to a different node, its inventory is synced with the new node, but no alerts are generated during this initial sync.
-  Content update: When content changes, all agents are re-scanned to ensure results are up to date, but no alerts are generated as part of this process.

 You can refer to the diagram below for a visual representation of this workflow.
```

**Review by sebasfalcone — COMMENTED** [review]
Some suggestions

**Comment by pereyra-m** [comment]
@sebasfalcone 
I've applied the suggestions

**Review by sebasfalcone — APPROVED** [review]

**Review by sebasfalcone — DISMISSED** [review]
LGTM

**Review by javimed — APPROVED** [review]

File: source/images/manual/vuln-detector/vuln-detector-workflow.png
[the remark this answers is not in the document]
Comment (sebasfalcone) [reply source/images/manual/vuln-detector/vuln-detector-workflow.png:1]: I agree with this

**Comment by sebasfalcone** [comment]
## Updated diagram
Vulnerability-scanner-alert-generation-ver2

**Review by javimed — APPROVED** [review]

**Pull request merged**